vir/src/callable_idents.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{ExprGen, PredicateAppGen, Type, PredicateAppGenData, StmtGenData, MethodCallGenData, VirCtxt};
+use private::*;
+
+#[derive(Debug, Clone, Copy)]
+pub struct FunctionIdent<'vir, A: Arity>(&'vir str, A);
+impl<'vir, T: Arity> FunctionIdent<'vir, T> {
+    pub const fn new(name: &'vir str, args: T) -> Self {
+        Self(name, args)
+    }
+    pub fn name(&self) -> &'vir str {
+        self.0
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct MethodIdent<'vir, A: Arity>(&'vir str, A);
+impl<'vir, T: Arity> MethodIdent<'vir, T> {
+    pub const fn new(name: &'vir str, args: T) -> Self {
+        Self(name, args)
+    }
+    pub fn name(&self) -> &'vir str {
+        self.0
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct PredicateIdent<'vir, A: Arity>(&'vir str, A);
+impl<'vir, T: Arity> PredicateIdent<'vir, T> {
+    pub const fn new(name: &'vir str, args: T) -> Self {
+        Self(name, args)
+    }
+    pub fn name(&self) -> &'vir str {
+        self.0
+    }
+}
+
+/// Module to prevent anyone else from implementing Arity
```

Comment:
Should we just use [`sealed`](https://docs.rs/sealed/latest/sealed/)?

Reply:
Done